core/contracts/oracle/implementation/Governor.sol

```diff
@@ -78,9 +79,10 @@ contract Governor is MultiRole, Testable {
      * disallows structs arrays to be passed to external functions.
      * @param transactions array of `Transaction` which can be voted on.
      */
-    function propose(Transaction[] memory transactions) public onlyRoleHolder(uint(Roles.Proposer)) {
+    function propose(Transaction[] memory transactions) public payable onlyRoleHolder(uint(Roles.Proposer)) {
```

Comment:
What happens if this proposal gets voted down?

Reply:
Great question had not thought about this. We could add another function like `withdrawProposal` which could allow the sender to withdraw their deposited ETH if and only if the proposal is voted down.

WDYT

Reply:
Hmmmm, that could work. Although that seems like a lot of complexity to add just for handling the ETH edge case that we don't really intend to use.

Here's an alternative that's admittedly less elegant: what if we just made the execute function payable and have it check that the `msg.value == transaction.value`? Is there anything wrong with not guaranteeing that the money's there before the proposal is voted on?

Reply:
>Here's an alternative that's admittedly less elegant: what if we just made the execute function payable and have it check that the msg.value == transaction.value? Is there anything wrong with not guaranteeing that the money's there before the proposal is voted on?

I think this is a reasonable solution. We could also make `executeProposal` pay back the original proposer if the following check fails: 

```
require(price != 0, "Cannot execute, proposal was voted down");
```

Reply:
After thinking about this more, I think the solution you proposed is actually more elegant as it does not require storing extra data. It just places the burden on the party interested in executing the proposal for depositing the correct amount of ETH

Reply:
That seems reasonable to me. I guess the only downside is that even if the transaction is voted through, it's not guaranteed to be executable by *anyone*.

Reply:
Presumably the original proposer would be interested in footing the bill to execute the proposal, otherwise yeah no one else would want to pay for it